### DISCUSSION TITLE
Cannot deploy the app from github using  Vercel REST API

### DISCUSSION THREAD

**Discussion by ghubnab99** [vercel/community#3402] [Feature Question]
### Question

We want to deploy an app from a connected github repo by invoking the Vercel Rest API: 
POST: Create a new deployment: (https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment)
https://api.vercel.com/v13/deployments

We were able to successfully create a new project through the rest api for the connected gihub repo, but are not able to create a new deployment.

We are not sure, if we're not getting the concepts of the projects vs deployments correctly.

Can we create a project and then deployment simultaneously in one single call through any specific API Endpoint.

Furthermore, the API Reference also does not have any examples or detailed information on the API params

Kindly requesting if anyone could please guide us in the correct direction.

Here's what we're doing:

POST: https://api.vercel.com/v13/deployments
(Auth Bearer Token is already being attached to the POST Request)

NOTE that we want to deploy the app from a connect github repo

{
    "branch": {
      "project_id": "prj_ODylioBmMtTvWRk9Qvqt3ZvQKYKV",
      "name": "new-deployment",
      "current_state": "init",
      "primary": true
    },
    "name": "my-instant-deployment",
    "framework": "nextjs"
}

Response: 400 Bad Request

{
    "error": {
        "code": "bad_request",
        "message": "Invalid request: \"files\" field should be an array"
    }
}

Thank you

**Comment by amyegan** [comment]
Duplicate of https://github.com/vercel/community/discussions/3403
